sdks/python/apache_beam/dataframe/frames.py

```diff
@@ -3037,10 +3059,8 @@ def do_partition_apply(df):
   tail = frame_base.wont_implement_method(
       DataFrameGroupBy, 'tail', explanation=_PEEK_METHOD_EXPLANATION)
 
-  first = frame_base.wont_implement_method(
-      DataFrameGroupBy, 'first', reason='order-sensitive')
-  last = frame_base.wont_implement_method(
-      DataFrameGroupBy, 'last', reason='order-sensitive')
+  first = frame_base.not_implemented_method('first')
```

Comment:
What's the difference here? I was looking at adjacent lines to see if there were any analogous functions, to understand at a high level, but I don't see any.

Reply:
`not_implemented_method` indicates a method that's not implemented just because we haven't gotten to it yet, it will `raise NotImplementedError(..)` if used. `wont_implement_method` is for operations that aren't implemented because of some structural issue (like being sensitive to order, or producing an output schema that we can't determine at construction time). The latter raises an error that will point to documentation about that type of limitation (BEAM-12029 for the error messages,  BEAM-11951 is for the documentation, that's still in progress). 

"Wont implement" is a little strong, since in practice we may still implement some of those in the future. But the barrier for those is higher.